packages/gatsby/src/internal-plugins/query-runner/page-query-runner.js

```diff
@@ -65,6 +70,12 @@ emitter.on(`DELETE_NODE`, action => {
   queuedDirtyActions.push({ payload: action.payload })
 })
 
+emitter.on(`CREATE_PAGE`, action => {
+  if (action.contextModified) {
+    queuedDirtyPathnames.push(action.payload.path)
```

Comment:
I think we can just reuse `runQueriesForPathnamesQueue` Map instead of creating new variable? We don't track why exactly query for pathname is dirty, so it should be safe to do

Reply:
Good suggestion! Thanks!
Addressed